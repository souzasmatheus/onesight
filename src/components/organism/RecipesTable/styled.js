import styled from 'styled-components';
import TableContainer from '@material-ui/core/TableContainer';

export const StyledTableContainer = styled(TableContainer)`
  margin: 20px auto 0;
  &.MuiTableContainer-root {
    max-width: 550px;
  }
`;
