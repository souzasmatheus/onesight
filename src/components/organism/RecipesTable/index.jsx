import React from 'react';
import { useSelector } from 'react-redux';

import Table from '@material-ui/core/Table';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableBody from '@material-ui/core/TableBody';
import Paper from '@material-ui/core/Paper';

import { RecipeInfo } from '~/components/molecules';

import { recipesSelector } from '~/feature/recipes/RecipesSlice';

import { StyledTableContainer } from './styled';

const RecipesTable = () => {
  const { recipes } = useSelector(recipesSelector);

  return (
    <StyledTableContainer component={Paper}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell />
            <TableCell align="center">Recipe</TableCell>
            <TableCell align="center">Category</TableCell>
            <TableCell align="center">Watch now!</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {recipes?.map((recipe) => (
            <RecipeInfo
              src={recipe?.strMealThumb}
              name={recipe.strMeal}
              category={recipe.strCategory}
              link={recipe?.strYoutube}
              description={recipe?.strInstructions}
            />
          ))}
        </TableBody>
      </Table>
    </StyledTableContainer>
  );
};

export default RecipesTable;
