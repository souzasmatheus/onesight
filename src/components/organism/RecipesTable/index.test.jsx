import React from 'react';
import { render } from '@testing-library/react';

import RecipesTable from './index';

import { MockedStore } from '~/../test/mocks/providers';

describe('<RecipesTable />', () => {
  let state;

  beforeEach(
    () =>
      (state = {
        recipes: [
          {
            strMeal: 'Arrabiata',
            strCategory: 'Italian',
          },
        ],
      }),
  );

  it('should match snapshot', () => {
    const tree = render(
      <MockedStore {...{ state }}>
        <RecipesTable />
      </MockedStore>,
    );

    expect(tree).toMatchSnapshot();
  });
});
