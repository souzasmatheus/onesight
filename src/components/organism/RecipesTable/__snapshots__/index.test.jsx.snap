// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesTable /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiTableContainer-root sc-kEjbxe gFRDUM MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              />
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Recipe
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Category
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                scope="col"
              >
                Watch now!
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiTableContainer-root sc-kEjbxe gFRDUM MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Recipe
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Category
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Watch now!
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
